src/main/webapp/app/shared/pipes/artemis-duration-from-seconds.pipe.ts

```diff
@@ -1,24 +1,109 @@
+import { Subscription } from 'rxjs';
+
+@Pipe({
+    name: 'artemisDurationFromSeconds',
+    pure: false,
+})
+export class ArtemisDurationFromSecondsPipe implements PipeTransform, OnDestroy {
+    private readonly secondsInDay = 60 * 60 * 24;
+    private readonly secondsInHour = 60 * 60;
+    private readonly secondsInMinute = 60;
+
+    private seconds: number;
+    private locale: string;
+    private localizedDuration: string;
+    private onLangChange?: Subscription;
+
+    constructor(private translateService: TranslateService) {}
 
-@Pipe({ name: 'artemisDurationFromSeconds' })
-export class ArtemisDurationFromSecondsPipe implements PipeTransform {
     /**
-     * Convert seconds to a human-readable duration format (mm:ss).
+     * Convert seconds to a human-readable duration format "d day(s) hh:mm::ss".
+     * The days and hours are left out if their value is zero
      * @param seconds {number}
      */
     transform(seconds: number): string {
-        const minutes = Math.floor(seconds / 60);
-        seconds = seconds - minutes * 60;
+        this.seconds = seconds;
+
+        const days = Math.floor(seconds / this.secondsInDay);
+        const hours = Math.floor((seconds % this.secondsInDay) / this.secondsInHour);
+        const minutes = Math.floor((seconds % this.secondsInHour) / this.secondsInMinute);
+        seconds = seconds % this.secondsInMinute;
+
+        let timeString = this.transformDays(days);
+
+        if (days > 0 || hours > 0) {
+            timeString += this.addLeadingZero(hours) + ':';
+        }
+
+        timeString += this.addLeadingZero(minutes) + ':';
+        timeString += this.addLeadingZero(seconds);
+
+        // Clean up existing subscription to onLangChange and subscribe to onLangChange event, in case the language changes.
+        this.cleanUpSubscription();
+        if (!this.onLangChange) {
+            this.onLangChange = this.translateService.onLangChange.subscribe((event: LangChangeEvent) => this.updateLocale(event.lang));
+        }
+
+        return timeString;
+    }
+
+    private addLeadingZero(number: number): string {
+        let numberOut = number.toString();
+        if (number < 10) {
+            numberOut = '0' + numberOut;
+        }
+        return numberOut;
+    }
+
+    private transformDays(days: number): string {
+        if (days === 0) {
+            return '';
+        }
+        // Set locale to current language.
+        this.updateLocale(this.translateService.currentLang);
+
+        return days + this.getDayString(days);
+    }
+
+    private getDayString(days: number): string {
+        if (this.locale === 'de') {
+            if (days > 1) {
+                return ' Tage ';
+            } else {
+                return ' Tag ';
+            }
+        } else {
+            if (days > 1) {
+                return ' days ';
+            } else {
+                return ' day ';
+            }
+        }
```

Comment:
```suggestion
        if (this.locale === 'de') {
            return days > 1 ? ' Tage ' : ' Tag ';
        } else {
            return days > 1 ? ' days ' : ' day ';
        }
```
Small suggestion to shorten the code a bit